app/controllers/api/v2/hosts_controller.rb

```diff
@@ -126,6 +126,13 @@ def destroy
         process_response @host.destroy
       end
 
+      api :GET, "/hosts/:id/enc", N_("Get ENC values of host")
+      param :id, :identifier_dottable, :required => true
+
+      def enc
+        render :json => { :data => @host.info }.to_json if @host
```

Comment:
Also the `if @host` is redundant I think, you already specify the parameter as `:required => true` above